### PR TITLE
Add System.Text.Json converter to models

### PR DESCRIPTION
**Pull request by ShivangiReja**
Fixes: https://github.com/Azure/autorest.csharp/issues/947

### CODE REVIEW

File: src/AutoRest.CSharp/Generation/Writers/SerializationWriter.cs [hunk 1]
```diff
@@ -104,6 +106,30 @@ private void WriteObjectSerialization(CodeWriter writer, ObjectType model)
...
+                    {
+                        WriteCustomJsonConverter(model, writer);
+                    }
+                }
+            }
+        }
+
+        private void WriteCustomJsonConverter(ObjectType model, CodeWriter writer)
+        {
+            writer.Append($"internal partial class {model.Declaration.Name}Converter : {typeof(JsonConverter)}<{model.Declaration.Name}>");
```
Comment (pakrym) [inline src/AutoRest.CSharp/Generation/Writers/SerializationWriter.cs:120]: `<{model.Declaration.Name}>")` use model.Type here.

File: test/TestProjects/ModelWithConverterUsage/Generated/Models/ModelStruct.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+#nullable disable
+
+using System;
+using System.Text.Json.Serialization;
+
+namespace ModelWithConverterUsage.Models
+{
+    /// <summary> The ModelStruct. </summary>
+    [JsonConverter(typeof(ModelStructConverter))]
```
Comment (pakrym) [inline test/TestProjects/ModelWithConverterUsage/Generated/Models/ModelStruct.cs:14]: Should we move this attribute to the other part of the file? The same where ModelStructConverter is?

File: src/AutoRest.CSharp/Output/Models/Types/ObjectType.cs [hunk 3]
```diff
@@ -86,6 +86,8 @@ public ObjectType(ObjectSchema objectSchema, BuildContext context): base(context
 
         public ObjectTypeProperty[] Properties => _properties ??= BuildProperties().ToArray();
 
+        public SchemaTypeUsage SchemaTypeUsage => _usage;
```
Comment (pakrym) [inline src/AutoRest.CSharp/Output/Models/Types/ObjectType.cs:89]: I wonder if we should put this value onto the `JsonSerialization` inside the `        public ObjectSerialization[] Serializations => _serializations ??= BuildSerializations();
`

Comment (pakrym) [reply src/AutoRest.CSharp/Output/Models/Types/ObjectType.cs:89]: Or better follow the pattern of IncludeSerializer method https://github.com/Azure/autorest.csharp/blob/4c0f9cf31af56decb03c67fdf99eb516c8ae81ae/src/AutoRest.CSharp/Output/Models/Types/ObjectType.cs#L306-L307

**Review by pakrym — APPROVED** [review]

File: test/TestProjects/ModelWithConverterUsage/Generated/Models/ModelStruct.cs (hunk 2 above)
Comment (ShivangiReja) [reply test/TestProjects/ModelWithConverterUsage/Generated/Models/ModelStruct.cs]: Yeah, this looks better.

**Review by pakrym — APPROVED** [review]

**Pull request merged**